src/main.rs

```diff
@@ -47,34 +47,37 @@ fn get_registry() -> Vec<(Principal, Vec<u64>)> {
 }
 
 #[query]
-fn tokens(user: Principal) -> Vec<u64> {
+fn soulbound_tokens(user: Principal) -> Vec<u64> {
     State::read_state(|state| state.get_user_registry(user))
 }
 
 #[update(guard = "is_admin")]
-fn add_asset(asset: Vec<u8>) -> Result<(), String> {
+fn add_asset(asset_name: String, asset: Vec<u8>) -> Result<(), String> {
     State::mutate_state(|state| {
-        state.set_image(asset);
+        state.set_image(asset_name, asset);
         Ok(())
     })
 }
 
 #[query]
-fn get_asset() -> Vec<u8> {
-    State::read_state(|state| state.get_image().to_vec())
+fn get_asset(asset_name: String) -> Result<Vec<u8>, String> {
+    State::read_state(|state| match state.get_image(&asset_name) {
+        Ok(bytes) => Ok(bytes.to_vec()),
+        Err(e) => Err(e),
+    })
 }
 
 #[update(guard = "is_admin")]
-fn clear_asset() {
+fn clear_asset(asset_name: String) {
     State::mutate_state(|state| {
-        state.clear_image();
+        state.clear_image(asset_name);
     })
 }
 
 #[update(guard = "is_admin")]
-fn append_asset(mut asset: Vec<u8>) -> Result<(), String> {
+fn append_asset(asset_name: String, mut asset: Vec<u8>) -> Result<(), String> {
```

Comment:
we're disabling this right or leaving it in?

Reply:
Not sure honestly.  For now the only way I have to test is through the CLI, which cannot push all the bytes at once (bash complains because I go over the char limit, so had to batch it)  Thought it might be useful later if we have ultra sized pngs?  But maybe not useful.  I'm ambivalent about it for now